src/libpriv/rpmostree-scripts.c

```diff
@@ -189,15 +189,44 @@ run_script_in_bwrap_container (int rootfs_fd,
   bwrap = rpmostree_bwrap_new (rootfs_fd, RPMOSTREE_BWRAP_MUTATE_ROFILES, error,
                                /* Scripts can see a /var with compat links like alternatives */
                                "--ro-bind", "./var", "/var",
-                               /* But no need to access persistent /tmp, so make it /tmp */
-                               "--bind", "/tmp", "/var/tmp",
                                /* Allow RPM scripts to change the /etc defaults; note we use bind
                                 * to ensure symlinks work, see https://github.com/projectatomic/rpm-ostree/pull/640 */
                                "--bind", "./usr/etc", "/etc",
                                NULL);
   if (!bwrap)
     goto out;
 
+  /* We just did a ro bind mount over /var above.  However we want a writable var/tmp,
+   * so we need to bind on top of it.
+   */
+  { struct stat stbuf;
+
+    /* Currently for most Atomic Host systems, we use sysroot/tmp.  However,
+     * I'd like to move away from this someday, since the "real" tmp feels squirreled
+     * away, and the symlink nature of it has broken systemd in the past.  Let's be
+     * robust here and support making it a plain old directory too.
+     */
+    if (fstatat (rootfs_fd, "tmp", &stbuf, AT_SYMLINK_NOFOLLOW) == 0
+        && S_ISDIR (stbuf.st_mode))
+      {
+        rpmostree_bwrap_append_child_argv (bwrap, "--bind", "./tmp", "/var/tmp", NULL);
+      }
+    /* Okay, this is the current AH default.  We have a sysroot dir in the base tree,
+     * but we won't have sysroot/tmp since our /sysroot here isn't pointing to the *real*
+     * sysroot.  In the tree it's just an empty dir.  Hence, we make sysroot/tmp */
+    else if (fstatat (rootfs_fd, "sysroot", &stbuf, AT_SYMLINK_NOFOLLOW) == 0
+             && S_ISDIR (stbuf.st_mode))
+      {
+        rpmostree_bwrap_append_child_argv (bwrap, "--dir", "/sysroot/tmp", NULL);
```

Comment:
In other words, I'd keep the `--bind ./tmp /var/tmp` in the call to `new()` like in the first version of the patch (to ensure that `/var/tmp` is taken care of), and then here, you only need to keep this branch (to ensure `/tmp` itself is actually valid).